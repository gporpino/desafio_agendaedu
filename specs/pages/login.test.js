import React from 'react';
import { shallow, mount } from 'enzyme';

import toJson from 'enzyme-to-json';

import Login from '../../src/pages/login';

describe('<Login />', () => {
  describe('inputs', () => {
    let wrapper = null;
    beforeAll(() => {
      wrapper = shallow(<Login />);
    });

    it('has login input', () => {
      expect(toJson(wrapper.find('#emailInput'))).not.toBe(null);

      expect(wrapper).toMatchSnapshot();
    });

    it('has password input', () => {
      expect(toJson(wrapper.find('#passwordInput'))).not.toBe(null);

      expect(wrapper).toMatchSnapshot();
    });

    it('has login button', () => {
      expect(toJson(wrapper.find('#loginButton'))).not.toBe(null);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when navigate', () => {
    let wrapper = null;
    const mock = jest.fn(async () => {
      wrapper.instance().afterLogin('@123456');
    });
    const mockNavigate = jest.fn();

    beforeEach(() => {
      wrapper = mount(<Login />);
      wrapper.instance().login = mock;

      wrapper.setProps({
        navigation: {
          navigate: mockNavigate
        }
      });
      wrapper.instance().forceUpdate();
    });

    afterEach(() => {
      mock.mockClear();
      mockNavigate.mockClear();
    });

    it('calls login when press button', () => {
      const button = wrapper.find('#loginButton').first();
      button.instance().props.onPress();

      expect(mock).toHaveBeenCalled();
    });

    it('navigate to Main with valid attributes', async () => {
      const button = wrapper.find('#loginButton').first();
      await button.instance().props.onPress();

      expect(wrapper.instance().props.navigation.navigate).toHaveBeenCalledWith(
        'Events'
      );
    });

    it('stay with invalid attributes', () => {
      wrapper.instance().afterLogin = jest.fn();
      wrapper.instance().forceUpdate();
      const button = wrapper.find('#loginButton').first();
      button.instance().props.onPress();

      expect(
        wrapper.instance().props.navigation.navigate
      ).not.toHaveBeenCalled();
    });
  });
});
