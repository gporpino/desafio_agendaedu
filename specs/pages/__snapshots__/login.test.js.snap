// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> inputs has login button 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 3,
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 36,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#333333",
            "fontFamily": "sf-pro-display-bold",
            "fontSize": 36,
            "fontWeight": "bold",
          }
        }
      >
        FaÃ§a seu login ğŸ”‘
      </Styled(Text)>
    </Component>
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 17,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#666666",
            "fontFamily": "sf-pro-display-regular",
            "marginBottom": 8,
          }
        }
      >
        E-mail ou usuÃ¡rio
      </Styled(Text)>
      <Component
        sytle={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          id="emailInput"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#733DBE",
              "borderRadius": 5,
              "borderWidth": 1,
              "color": "#333333",
              "fontFamily": "sf-pro-display-regular",
              "height": 50,
              "paddingBottom": 13,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 13,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Icon
          allowFontScaling={false}
          name="email-outline"
          size={24}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#AAAAAA",
              "marginRight": 15,
              "marginTop": -37,
            }
          }
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 17,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#666666",
            "fontFamily": "sf-pro-display-regular",
            "marginBottom": 8,
          }
        }
      >
        Senha
      </Styled(Text)>
      <Component
        sytle={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          id="passwordInput"
          onChangeText={[Function]}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#ABB1B7",
              "borderRadius": 5,
              "borderWidth": 1,
              "color": "#333333",
              "fontFamily": "sf-pro-display-regular",
              "height": 50,
              "paddingBottom": 13,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 13,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Icon
          allowFontScaling={false}
          name="eye-off-outline"
          size={24}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#AAAAAA",
              "marginRight": 15,
              "marginTop": -37,
            }
          }
        />
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "alignSelf": "stretch",
        "padding": 30,
      }
    }
  >
    <Styled(Button)
      id="loginButton"
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#733DBE",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 40,
          "paddingBottom": 16,
          "paddingTop": 16,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "fontWeight": "600",
          }
        }
      >
        Entrar
      </Styled(Text)>
    </Styled(Button)>
  </Component>
</Component>
`;

exports[`<Login /> inputs has login input 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 3,
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 36,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#333333",
            "fontFamily": "sf-pro-display-bold",
            "fontSize": 36,
            "fontWeight": "bold",
          }
        }
      >
        FaÃ§a seu login ğŸ”‘
      </Styled(Text)>
    </Component>
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 17,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#666666",
            "fontFamily": "sf-pro-display-regular",
            "marginBottom": 8,
          }
        }
      >
        E-mail ou usuÃ¡rio
      </Styled(Text)>
      <Component
        sytle={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          id="emailInput"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#733DBE",
              "borderRadius": 5,
              "borderWidth": 1,
              "color": "#333333",
              "fontFamily": "sf-pro-display-regular",
              "height": 50,
              "paddingBottom": 13,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 13,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Icon
          allowFontScaling={false}
          name="email-outline"
          size={24}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#AAAAAA",
              "marginRight": 15,
              "marginTop": -37,
            }
          }
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 17,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#666666",
            "fontFamily": "sf-pro-display-regular",
            "marginBottom": 8,
          }
        }
      >
        Senha
      </Styled(Text)>
      <Component
        sytle={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          id="passwordInput"
          onChangeText={[Function]}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#ABB1B7",
              "borderRadius": 5,
              "borderWidth": 1,
              "color": "#333333",
              "fontFamily": "sf-pro-display-regular",
              "height": 50,
              "paddingBottom": 13,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 13,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Icon
          allowFontScaling={false}
          name="eye-off-outline"
          size={24}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#AAAAAA",
              "marginRight": 15,
              "marginTop": -37,
            }
          }
        />
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "alignSelf": "stretch",
        "padding": 30,
      }
    }
  >
    <Styled(Button)
      id="loginButton"
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#733DBE",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 40,
          "paddingBottom": 16,
          "paddingTop": 16,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "fontWeight": "600",
          }
        }
      >
        Entrar
      </Styled(Text)>
    </Styled(Button)>
  </Component>
</Component>
`;

exports[`<Login /> inputs has password input 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 3,
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 36,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#333333",
            "fontFamily": "sf-pro-display-bold",
            "fontSize": 36,
            "fontWeight": "bold",
          }
        }
      >
        FaÃ§a seu login ğŸ”‘
      </Styled(Text)>
    </Component>
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 17,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#666666",
            "fontFamily": "sf-pro-display-regular",
            "marginBottom": 8,
          }
        }
      >
        E-mail ou usuÃ¡rio
      </Styled(Text)>
      <Component
        sytle={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          id="emailInput"
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "#733DBE",
              "borderRadius": 5,
              "borderWidth": 1,
              "color": "#333333",
              "fontFamily": "sf-pro-display-regular",
              "height": 50,
              "paddingBottom": 13,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 13,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Icon
          allowFontScaling={false}
          name="email-outline"
          size={24}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#AAAAAA",
              "marginRight": 15,
              "marginTop": -37,
            }
          }
        />
      </Component>
    </Component>
    <Component
      style={
        Object {
          "alignSelf": "stretch",
          "marginBottom": 17,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "color": "#666666",
            "fontFamily": "sf-pro-display-regular",
            "marginBottom": 8,
          }
        }
      >
        Senha
      </Styled(Text)>
      <Component
        sytle={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          id="passwordInput"
          onChangeText={[Function]}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#ABB1B7",
              "borderRadius": 5,
              "borderWidth": 1,
              "color": "#333333",
              "fontFamily": "sf-pro-display-regular",
              "height": 50,
              "paddingBottom": 13,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 13,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Icon
          allowFontScaling={false}
          name="eye-off-outline"
          size={24}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#AAAAAA",
              "marginRight": 15,
              "marginTop": -37,
            }
          }
        />
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "alignSelf": "stretch",
        "padding": 30,
      }
    }
  >
    <Styled(Button)
      id="loginButton"
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#733DBE",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 40,
          "paddingBottom": 16,
          "paddingTop": 16,
        }
      }
    >
      <Styled(Text)
        style={
          Object {
            "fontWeight": "600",
          }
        }
      >
        Entrar
      </Styled(Text)>
    </Styled(Button)>
  </Component>
</Component>
`;
